From 5c0cf67aafe54a31e1e2dc64a0309b0d4a70beb5 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Thu, 29 Sep 2022 14:58:10 +0200
Subject: [PATCH 0246/1141] remoteproc: stm32: set the sysfs firmware image
 format

Set the value of the firmware image format for the sysfs API depending
on the delegation to the TEE for loading a signed firmware.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I7263ea35e46231f4b73c90db12573fe97ecbe77d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/268346
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 drivers/remoteproc/stm32_rproc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/remoteproc/stm32_rproc.c
+++ b/drivers/remoteproc/stm32_rproc.c
@@ -1048,8 +1048,12 @@ static int stm32_rproc_probe(struct plat
 	ddata = rproc->priv;
 	ddata->desc = desc;
 	ddata->trproc = trproc;
-	if (trproc)
+	if (trproc) {
 		ddata->trproc->rproc = rproc;
+		rproc->fw_format = RPROC_FW_TEE;
+	} else {
+		rproc->fw_format = RPROC_FW_ELF;
+	}
 
 	rproc_coredump_set_elf_info(rproc, ELFCLASS32, EM_NONE);
 
